File: LLM-Translations/GPT-4/codenet/Java/C/s453440639.c
#include <stdio.h>

int main() {
    int H, W;
    scanf("%d %d", &H, &W);
    getchar();
    char S[H][W];
    
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            scanf("%c", &S[i][j]);
        }
        getchar();
    }

    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            if (S[i][j] == '.') {
                int count = 0;
                for (int a = -1; a <= 1; a++) {
                    if (i + a < 0 || H <= i + a) {
                        continue;
                    }
                    for (int b = -1; b <= 1; b++) {
                        if (j + b < 0 || W <= j + b) {
                            continue;
                        }
                        if ( S[i + a][j + b] == '#') {
                            count++;
                        }
                    }
                }
                S[i][j] = (char)(count+48);
            }
            printf("%c", S[i][j]);
        }
        printf("\n");
    }

    return 0;
}
// 

File: LLM-Translations/GPT-4/codenet/Java/Go/s453440639.go
package main

import (
	"fmt"
	"strings"
	"bufio"
	"os"
)

func main() {
	scanner := bufio.NewReader(os.Stdin)

	var H, W int
	fmt.Scanf("%d %d\n", &H, &W)

	S := make([]string, H)

	for i := 0; i < H; i++ {
		S[i], _ = scanner.ReadString('\n')
		S[i] = strings.TrimSpace(S[i])
	}

	for i := 0; i < H; i++ {
		for j := 0; j < W; j++ {
			if S[i][j] == '.' {
				count := 0
				for a := -1; a <= 1; a++ {
					if i + a < 0 || H <= i + a {
						continue
					}
					for b := -1; b <= 1; b++ {
						if j + b < 0 || W <= j + b {
							continue
						}
						if S[i + a][j + b] == '#' {
							count++
						}
					}
				}
				S[i] = S[i][:j] + string(count+48) + S[i][j+1:]
			}
			fmt.Printf("%c", S[i][j])
		}
		fmt.Printf("\n")
	}
}
// 
